### PR TITLE
PR pipeline: collapse pipeline into one job

### PR DESCRIPTION
**Pull request by jingyimei**
We do this to ensure all PRs will get picked up by the pipeline. Because
of a Concourse bug with specifying `version: every`, we are not able to
have a conventional pipeline with separate jobs.

Signed-off-by: Jingyi Mei <jmei@pivotal.io>

### CODE REVIEW

File: concourse/pipelines/pr_pipeline.yml [hunk 1]
```diff
@@ -24,231 +24,119 @@ resources:
...
-        TARGET_OS_VERSION: 6
-        BLD_TARGETS: ""
-      on_failure:
-        put: gpdb_pr
+      - task: compile_gpdb_open_source_centos6
+        file: gpdb_pr/concourse/tasks/compile_gpdb_open_source.yml
+        image: centos-gpdb-dev-6
+        input_mapping:
+          gpdb_src: gpdb_pr
+        on_failure:
```
Comment (ashwinstar) [inline concourse/pipelines/pr_pipeline.yml]: Can you define this part as variable (I do not know right xml term for it) as same repeated many times below.

File: concourse/pipelines/pr_pipeline.yml [hunk 2]
```diff
@@ -24,231 +24,119 @@ resources:
...
-        - compile_gpdb_open_source_centos6
-      - get: bin_gpdb
-        resource: bin_gpdb_centos
-        passed: [compile_gpdb_centos6]
-        version: every
-        trigger: true
-      - get: centos-gpdb-dev-6
-    - task: MU_check_centos
-      file: gpdb_pr/concourse/tasks/gpMgmt_check_gpdb.yml
+    - task: separate_qautils_files_for_rc
```
Comment (ashwinstar) [inline concourse/pipelines/pr_pipeline.yml]: Do we really need this, I don't think its required ?
PR pipeline is only used in my opinion to validate code and no output from it is consumed in any other form than boolean pass or fail.

File: concourse/pipelines/pr_pipeline.yml [hunk 3]
```diff
@@ -24,231 +24,119 @@ resources:
...
-      - get: centos-gpdb-dev-6
-      - get: gpaddon_src
-    - task: separate_qautils_files_for_rc
-      file: gpdb_pr/concourse/tasks/separate_qautils_files_for_rc.yml
-      image: centos-gpdb-dev-6
-      input_mapping:
-        gpdb_src: gpdb_pr
-      params:
-        QAUTILS_TARBALL: rc_bin_gpdb/QAUtils-rhel6-x86_64.tar.gz
     - task: gpdb_rc_packaging_centos
```
Comment (ashwinstar) [inline concourse/pipelines/pr_pipeline.yml]: Even think don't need this, unless someone justifies its need.

File: concourse/pipelines/pr_pipeline.yml (hunk 1 above)
Comment (tom-meyer) [inline concourse/pipelines/pr_pipeline.yml]: Nice idea. We should be able to use yaml references for the on_failure blocks. We'll try it out.

File: concourse/pipelines/pr_pipeline.yml (hunk 3 above)
Comment (tom-meyer) [inline concourse/pipelines/pr_pipeline.yml]: Same comment applied here.

File: concourse/pipelines/pr_pipeline.yml (hunk 2 above)
Comment (tom-meyer) [inline concourse/pipelines/pr_pipeline.yml]: We maintained the same jobs that existed in the previous (expanded) pipeline. Your team is welcome to make appropriate changes.

**Comment by cjcjameson** [comment]
Can you fly up a paused pipeline so that we can see how it will look? Exciting!

**Comment by tom-meyer** [comment]
@cjcjameson It is up and (currently) green! https://gpdb.ci.pivotalci.info/teams/gpdb/pipelines/gpdb_pr

**Pull request merged**